src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -802,55 +643,139 @@ private void VerifyCallAllowed(bool synchronousCall)
         }
 
         /// <summary>
-        /// Catch any exception thrown by the action passed in; in the exception case move the writer into
-        /// state ExceptionThrown and then rethrow the exception.
+        /// Verifies that calling CreateOperationRequestMessage is valid.
         /// </summary>
-        /// <param name="action">The action to execute.</param>
-        private void InterceptException(Action action)
+        /// <param name="synchronousCall">true if the call is to be synchronous; false otherwise.</param>
+        /// <param name="method">The Http method to be used for this request operation.</param>
+        /// <param name="uri">The Uri to be used for this request operation.</param>
+        /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
+        private void VerifyCanCreateOperationRequestMessage(bool synchronousCall, string method, Uri uri,
+            string contentId)
         {
-            try
+            // Common verification.
+            this.ValidateWriterReady();
+            this.VerifyCallAllowed(synchronousCall);
+
+            if (this.outputContext.WritingResponse)
             {
-                action();
+                this.ThrowODataException(Strings.ODataBatchWriter_CannotCreateRequestOperationWhenWritingResponse);
             }
-            catch
+
+            ExceptionUtils.CheckArgumentStringNotNullOrEmpty(method, "method");
+
+            ExceptionUtils.CheckArgumentNotNull(uri, "uri");

```

Comment:
Not a big deal, but these normally go first in the method.

Reply:
Make sense. Moved validations to beginning of method. 

---
In reply to: [145852397](https://github.com/OData/odata.net/pull/960#discussion_r145852397) [](ancestors = 145852397)